Engine/platform/windows/gfx/ali3dd3d.cpp

```diff
@@ -1815,15 +1815,17 @@ void D3DGraphicsDriver::do_fade(bool fadingOut, int speed, int targetColourRed,
 
   if (speed <= 0) speed = 16;
   speed *= 2;  // harmonise speeds with software driver which is faster
+
+  // INNER GAME LOOP - d3d fade
   for (int a = 1; a < 255; a += speed)
   {
     d3db->SetTransparency(fadingOut ? a : (255 - a));
     this->_renderAndPresent(flipTypeLastTime, false);
 
-    do {
-      if (_pollingCallback)
-        _pollingCallback();
-    } while (waitingForNextTick());
+    // GAME LOOP DELAY has occured by this point.
```

Comment:
Is this correct? Direct3D _renderAndPresent does not do anything besides actually rendering textures.